pylint/config/utils.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+import sys
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pylint import reporters
+from pylint.utils import utils
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+def _config_initialization(
+    linter: "PyLinter",
+    args_list: List[str],
+    reporter: Union[reporters.BaseReporter, reporters.MultiReporter, None] = None,
+    config_file: Optional[str] = None,
+    verbose_mode: Optional[bool] = None,
+) -> List[str]:
+    """Parse all available options, read config files and command line arguments and
+    set options accordingly.
+    """
+
+    # Read the config file. The parser is stored on linter.cfgfile_parser
```

Comment:
I does not make much sense, right ? Why isn't it in the ``pylint.config`` package ? (Refactor for later, or maybe even prior to this one ?)

Reply:
I'm not sure. But it is kinda nice: we have access to it everywhere we have access to the `linter`. (Exactly the opposite of the problem I'm facing with `argparse` right now). So for now I'd said let's leave it as is.